src/value_generators/test/DocumentGeneratorTestCases.yml

```diff
@@ -177,6 +183,19 @@ Tests:
           xYYXCh/Ybl: { "$numberLong": "9" }
           EHcmXZX/d4: { "$numberLong": "10" }
 
+  - Name: Object Generator disallow duplicate keys
+    GivenTemplate:
+      a:
+        {^Object: {
+          withNEntries: 3,
+          havingKeys: "c",
+          andValues: {^Inc: {}},
+          allowDuplicateKeys: false
```

Comment:
This test-case wasn't obvious to me at first. I think add a comment here similar to what you had below. Even though the template has `withNEntries: 3`, the generated document will only have one entry because the key for each entry was specified as a constant, and `allowDuplicateKeys: false` prevented additional (duplicate) entries from being added. Something like that.

Reply:
I also attempted to add a test that might be able to observe at least some difference in the behaviour between the two cases. I gave up on running the tests locally, so if it fails in evergreen I'll only keep the test with "skip" setting.